packages/state-transition/src/epoch/processPendingBalanceDeposits.ts

```diff
@@ -1,26 +1,49 @@
+import {FAR_FUTURE_EPOCH} from "@lodestar/params";
 import {CachedBeaconStateElectra} from "../types.js";
 import {increaseBalance} from "../util/balance.js";
 import {getActivationExitChurnLimit} from "../util/validator.js";
+import {getCurrentEpoch} from "../util/epoch.js";
 
 /**
  * Starting from Electra:
  * Process pending balance deposits from state subject to churn limit and depsoitBalanceToConsume.
  * For each eligible `deposit`, call `increaseBalance()`.
  * Remove the processed deposits from `state.pendingBalanceDeposits`.
  * Update `state.depositBalanceToConsume` for the next epoch
+ * 
+ * TODO Electra: Update ssz library to support batch push to `pendingBalanceDeposits`
  */
 export function processPendingBalanceDeposits(state: CachedBeaconStateElectra): void {
   const availableForProcessing = state.depositBalanceToConsume + BigInt(getActivationExitChurnLimit(state));
+  const currentEpoch = getCurrentEpoch(state);
   let processedAmount = 0n;
   let nextDepositIndex = 0;
+  const depositsToPostpone = [];
 
   for (const deposit of state.pendingBalanceDeposits.getAllReadonly()) {
-    const {amount} = deposit;
-    if (processedAmount + amount > availableForProcessing) {
-      break;
+    const {amount, index: depositIndex} = deposit;
+    const validator = state.validators.get(depositIndex);
```

Comment:
```suggestion
    const validator = state.validators.getReadOnly(depositIndex);
```

Reply:
This is fixed in #6892 to pass related spec tests 👍